geopandas/geodataframe.py

```diff
@@ -575,6 +576,31 @@ def merge_geometries(block):
 
         return aggregated
 
+    # overrides GeoPandasBase method
+    def explode(self):
+        """
+        Explode multi-part geometries into multiple single geometries.
+
+        Single rows can become multiple rows.
+        This is analogous to PostGIS's ST_Dump(). 
+
+        Returns
+        ------
+        A GeoDataFrame with rows containing MultiPoints expanded into single
+        Point rows. The rest of the original columns are preserved.
+        """
+        exploded_geom = self.geometry.explode().reset_index()
+        exploded_geom.rename(columns={0:'geometry'})
+        df = exploded_geom.merge(
+            self.drop('geometry', axis=1),
+            left_on='level_0',
+            right_index=True)
+        df.set_index(['level_0', 'level_1'], inplace=True)
```

Comment:
If the original dataframe has a named index, this will not work I think. Can you test it with a dataframe with an index name?

Reply:
Yup will do. Sorry this is taking so long!